invenio_records_resources/services/records/service.py

```diff
@@ -579,3 +581,85 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def create_or_update_many(self, identity, data, uow=None):
+        """Create or update a list of records.
+
+        This method takes a list of record data and creates or updates the corresponding records.
+
+        :param identity: The user identity performing the operation.
+        :param data: A list of tuples containing the record ID and record data.
+        :param uow: The unit of work to register the record operations. Defaults to None.
+        """
+        records_processed = []
+
+        def _create_record(record_dict, record):
+            """Create a record."""
+            # Permissions
+            self.require_permission(identity, "update", record=record)
+            record_data, schema_errors = self.schema.load(
+                record_dict,
+                context=dict(identity=identity, pid=record.pid, record=record),
+                raise_errors=False,
+            )
+
+            # If errors we avoid creating the record
+            if schema_errors:
+                records_processed.append(("update", record_dict, schema_errors, None))
+                return
+
+            # Run components
+            self.run_components(
+                "update", identity, data=record_data, record=record, uow=uow
+            )
+
+            records_processed.append(("update", record, schema_errors, None))
```

Comment:
is it on purpose that the "update" key doesn't match the function name `_create_record`. same for `def _update_record`

Reply:
Yes, the permissions are the same as for `update` and `create`. Ideally we would want to have one permission check at the `create_update_many` (also to be more efficient) but it's not easy at the moment as for update we need to pass the record and for create we don't. Something to improve for the future.

Reply:
for me it is counterintuitive that the `update` keyword is used in the in the `_create_record` function. the name of the function is `_create_record` the docstring says it creates records but the the `run_components` call uses `update`

when you have a look to the `def _create` function you see that the for the [permission call](https://github.com/inveniosoftware/invenio-records-resources/blob/cefb51669bf5aa98f92631c098a439092f3bb168/invenio_records_resources/services/records/service.py#L341) for the [run_components call](https://github.com/inveniosoftware/invenio-records-resources/blob/cefb51669bf5aa98f92631c098a439092f3bb168/invenio_records_resources/services/records/service.py#L356) `create` as keyword is used.

Reply:
For the permissions checks maybe it makes sense that we just do the one check in the beginning of the `create_or_update_many` service call for a `can_create_or_update_many`? It should only allow for the system identity.

Reply:
Done, added https://github.com/inveniosoftware/invenio-records-permissions/pull/100

Reply:
Merged and released: https://pypi.org/project/invenio-records-permissions/0.20.0/